colossalai/communication/collective.py

```diff
@@ -54,6 +59,7 @@ def reduce_scatter(tensor: Tensor,
                    dim: int,
                    parallel_mode: ParallelMode,
                    op: ReduceOp = ReduceOp.SUM,
+                   on_cpu: bool = False,
```

Comment:
on_cpu may have a conflict with tensor.device.
we have to ensure tensor.device is CPU?